extensions/panache/panache-common/deployment/src/main/java/io/quarkus/panache/common/deployment/PanacheEntityEnhancer.java

```diff
@@ -80,15 +82,26 @@ public FieldVisitor visitField(int access, String name, String descriptor, Strin
                 @Override
                 public AnnotationVisitor visitAnnotation(String descriptor, boolean visible) {
                     descriptors.add(descriptor);
-                    return super.visitAnnotation(descriptor, visible);
+                    if (!descriptor.startsWith(JAXB_ANNOTATION_PREFIX)) {
+                        return super.visitAnnotation(descriptor, visible);
+                    }
+                    // Save off JAX-B annotations on the field so they can be applied to the generated getter later
+                    EntityFieldAnnotation efAnno = new EntityFieldAnnotation(descriptor);
+                    ef.annotations.add(efAnno);
+                    return new AnnotationVisitor(Opcodes.ASM7) {
```

Comment:
I forgot what the ASM API does with annotation visitors wrt nested annotations. Does the jaxb module have nested annotations? Are we sure that this won't be called for nested annotation attributes, or fail to properly forward those calls?

Reply:
you're correct, there are separate visit method to override for:
- nested annotation --> visitAnnotation
- enums --> visitEnum
- arrays --> visitArray (javadoc indicates it calls `visit()` for each array element)

I had a look through  the annotations in the `javax.xml.bind.annotation` package, and I see there are no annotations with enum or nested annotation attributes, but I do see there is a `XmlElements` and `XmlElementRefs` container annotations which take arrays of annotations. Looking back at the code I don't think these cases will be handled properly, so I'll add a test for it and ensure they work properly.

Reply:
@FroMage I pushed another commit which properly handles the `@XmlElements` and `@XmlElementRefs` scenario. Ready for re-review

Reply:
If you've checked the JAXB API and found only a single annotation whose value is an array of nested annotations, I'm torn between saying this is Good Enough™ or saying this has to be more general because if they ever add a new attribute with an enum value or a non-"value" non-array nested annotation, this code will suddenly break.

On the other hand, "collecting all annotations that match a given predicate and moving them to another element" sounds like a sufficiently reusable functionality that it might be worth doing this comprehensively into a quarkus-core utility class that others can later reuse.

WDYT @stuartwdouglas @dmlloyd?

Reply:
I think JAXB is rather unlikely to add new annotations in the future, what I'd be more concerned about is the impending `javax.*` --> `jakarta.*` package rename and handling that case properly. However, I've left that out because I assume some sort of general solution will be used for Quarkus and it may cover this scenario too (but I doubt it)

Reply:
I do wonder if there isn't already something in ASM that lets us "capture" a visitor's "visits" and apply it somewhere else. It feels like a problem that must happen pretty often.

The other thing when it comes to Good Enough™ is that you officially becomes the maintainer of that piece of code and by tradition you have to add a `// FIXME: this only works for XXX case but I'll fix it if it ever breaks` ;)

Reply:
Yea, recording the annotation processing flow and redirecting it to a different element was my preference as well, but I could not see a way to do this because in order to redirect the annotation processing we would need the Visitor instance from the generated accessor method so we could delegate to it. However, the visitor for the generated getter doesn't get created until later on in the code flow after we've processed all of the fields. So at this point it's a dead-end chicken-and-the-egg issue because we cannot delegate to the getter method visitor until after we've scanned the field and determined we need to generate a getter in the first place.

I'm pretty confident in the code and the test coverage (as long as the scope remains moving JAX-B annotations and not other annotations), so I could certainly add a `FIXME` to put myself on the hook for maintaining it in the future.

Reply:
OK, let's just leave a day for @stuartwdouglas or @dmlloyd to confirm they don't have or know of an API that already does this annotation visitor transfer, which would render the discussion moot, and then I'll approve this PR.

Let's also ask @geoand who also has some m4d 4SM ski11z

Reply:
I certainly don't, but let me read trough this and see if I have anything to suggest :)

Reply:
I don't know of any ASM API off hand for what you are looking for here.  The general solution @FroMage talks about would be great, but I would assume reading through the PR that this is just fine for the time being and perhaps we can reexamine in the future if need be (which I assume we won't have to :) )?

Reply:
OK, let's not wait for further input, this is fine, and we can revisit later if need be.